internal/policies/mount/mount.go

```diff
@@ -293,3 +503,50 @@ func (m *Manager) cleanupMountsFile(ctx context.Context, uid string) (err error)
 	}
 	return nil
 }
+
+// cleanupMountUnits removes all the mount units generated by adsys for the current system.
+func (m *Manager) cleanupMountUnits(ctx context.Context, units []string) (err error) {
+	defer decorate.OnError(&err, i18n.G("failed to clean up the mount units"))
+
+	for _, unit := range units {
+		// Stops and disables the unit before removing it
+		if err = m.execSystemCtlCmd(ctx, "disable", "--now", unit); err != nil {
```

Comment:
disable --now -> what happens if the stop fails, does it still disable the unit? (which is what we want, for next boot)

Reply:
It should. Since disable is not dependent on stop, I'm assuming the command disables the unit first and then tries to stop it. The documentation does not specify its behavior explicitly, but this section reinforces the thinking: 
```
Note that this command does not implicitly stop the units that are being disabled. If this is desired, either combine this command with the --now switch, or invoke the stop command with appropriate arguments later.
```

Reply:
Do you mind testing? :)

Reply:
Well, I'll do it once I figure out a good way to do it